spinflip/time_plots.py

```diff
@@ -6,15 +6,30 @@
 
 #generates plots vs time of spin parameters in fast flavor instability simulation
 class TimePlots:
-    def __init__(self, data_loc,  merger_data_loc, location, p_abs, resonance_type, initial_ket):
-        
-        self.data_loc = data_loc
-        self.h5file = h5py.File(self.data_loc, "r")
+    def __init__(self,
+                 it_lim, 
+                 emu_data_loc,
+                 merger_data_loc,
+                 location,
+                 p_abs, 
+                 resonance_type = 'simplified',
+                 initial_ket = np.array([1,0,0,0,0,0])):
+                
+        self.emu_data_loc = emu_data_loc
+        self.h5file = h5py.File(self.emu_data_loc, "r")
  
-        self.time_axis = np.array(self.h5file["t(s)"])*1E9 #ns
-        self.nt = self.time_axis.shape[0]
+        self.time_axis = np.array(self.h5file["t(s)"])[it_lim[0]:it_lim[1]]*1E9 #ns
+        self.nt = self.time_axis.shape[0]-1
```

Comment:
I think this is a confusing way to do things that will lead to indexing issues. The problem here is that the upper limit in python indexing is not inclusive, so the line above this should account for that. The number of times and the shape of the time array should match to avoid confusion.

Reply:
so basically instead of having nt be time_axis.shape[0] -1 we should have the upper limit be it_lim[1]+1 ?